pkg/apis/resource/extension_view.go

```diff
@@ -176,10 +168,21 @@ func (r *RouteUpgradeRequest) Validate() error {
 		r.ResourceDefinitionMatchingRule = &model.ResourceDefinitionMatchingRuleQueryInput{
 			ID: rule.ID,
 		}
+
+		en.ResourceDefinitionMatchingRuleID = &rule.ID
 	default:
 		return errors.New("template or resource definition is required")
 	}
 
+	if !r.ReuseAttributes {
+		computedAttr, err := pkgresource.GenComputedAttributes(r.Context, r.Client, en)
+		if err != nil {
+			return err
+		}
+
+		r.ComputedAttributes = computedAttr
+	}
+
```

Comment:
we should make `if-else` look like a whole logic, please put the above `if` block here too.

```
if r.ReuseAttributes {
		r.Attributes = entity.Attributes
		r.ComputedAttributes = entity.ComputedAttributes
		if r.Labels == nil {
			r.Labels = entity.Labels
		}
} else {
                computedAttr, err := pkgresource.GenComputedAttributes(r.Context, r.Client, en)
		if err != nil {
			return err
		}

		r.ComputedAttributes = computedAttr
}
```